modules/images/webp-uploads/load.php

```diff
@@ -573,3 +575,43 @@ function webp_uploads_update_featured_image( $html, $post_id, $attachment_id ) {
 	return webp_uploads_img_tag_update_mime_type( $html, 'post_thumbnail_html', $attachment_id );
 }
 add_filter( 'post_thumbnail_html', 'webp_uploads_update_featured_image', 10, 3 );
+
+/**
+ * Returns an array of image sizes that can have alternative mime types. By default, only core sizes
+ * are allowed and will be returned.
+ *
+ * If a custom size wants to have additional mime types support, they will need to have
+ * the provide_additional_mime_types property in the size details or it should be added using the
+ * webp_uploads_image_sizes_with_additional_mime_type_support filter.
+ *
+ * @since n.e.x.t
+ *
+ * @return array An array of image sizes that can have additional mime types.
+ */
+function webp_uploads_get_image_sizes_with_additional_mime_type_support() {
+	$additional_sizes = wp_get_additional_image_sizes();
+	$allowed_sizes    = array(
+		'thumbnail'      => 'thumbnail',
+		'medium'         => 'medium',
+		'medium_large'   => 'medium_large',
+		'large'          => 'large',
+		'post-thumbnail' => 'post-thumbnail',
+	);
+
+	foreach ( $additional_sizes as $size => $size_details ) {
+		if ( ! empty( $size_details['provide_additional_mime_types'] ) ) {
```

Comment:
where does `provide_additional_mime_types` property come from or this something you were thinking we would add later?

Reply:
This would be the field that can later be populated via `add_image_size`'s additional parameter. So having this here prepares us for that.
Technically, one could already set it now by directly touching the `$_wp_additional_image_sizes` global - which of course is discouraged, but I think it makes sense supporting this here already since we'll need it anyway and it's technically possible.